### PR TITLE
Create IErrorDetail.java

### PR DESCRIPTION
**Pull request by hnorrie16**

### CODE REVIEW

**Review by HarryGHeathcock — APPROVED** [review]
Looks good to me

File: src/main/java/io/electrum/vas/interfaces/IErrorDetail.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package io.electrum.vas.interfaces;
+
+public interface IErrorDetail {
```
Comment (TimElectrum) [inline src/main/java/io/electrum/vas/interfaces/IErrorDetail.java:3]: Please add some javadoc here to let others know in what version of the base API this was introduced. This is just ripe for issues where people _think_ they're using the interface but they're using the wrong version of the base API and then they have issues and can't track them down easily.

Comment (TimElectrum) [inline src/main/java/io/electrum/vas/interfaces/IErrorDetail.java:3]: Please add some javadoc here to let others know in what version of the base API this was introduced. This is just ripe for issues where people _think_ they're using the interface but they're using the wrong version of the base API and then they have issues and can't track them down easily.

**Review by TimElectrum — COMMENTED** [review]
This is simply a code change to `master` and not a distinguished release. Please avoid ad-hoc code changes to `master`. Every change to `master` should be versioned with an appropriate description of the change in the release notes:

- Please bump the version. Changes to `master` should be versioned.
- Please add an entry to the release notes describing the change. 

Another issue is the lack of testing. I'm not a fan of testing for testing's sake. 100% code coverage is a guide, not a goal by itself. But it's awfully easy for someone to build on this version of the base API (with this new interface) and just forget to use it. I think a test such as the following might be worthwhile:

1. Check if there is a class called `ErrorDetail` on the classpath.
2. If there is, make sure it implements the `IErrorDetail` interface.
3. If it doesn't implement the interface, fail the test.

If you add it to the `NewModelTests` then it should be inherited by our other APIs because their `NewModelTests` inherit from that. I'm just not sure whether the inherited tests are run or not so check that.

Lastly, when you merge this to `master`, please be sure to release it at the same time. Otherwise the risk is that there's an unreleased version. This isn't really a big issue but:

- It's just good practice to keep things tidy and haphazard releases isn't tidy.
- One day someone is going to investigate an issue related to this change and try to track down the version it was introduced in and if that version was never released it's going to be a massive red-herring.

**Pull request merged**